backend/src/main/scala/cromwell/backend/package.scala

```diff
@@ -35,6 +35,9 @@ package object backend {
                                   key: BackendJobDescriptorKey,
                                   inputs: Map[LocallyQualifiedName, WdlValue]) {
     val call = key.call
+
+    /** The abort functionality potentially becomes available after the job descriptor is created */
+    var abortFunction: Option[() => Unit] = None
```

Comment:
With the caveat that I haven't looked at this nearly as deeply as you have, would it be possible to have the JesJobExecutionActor send an abort to the JesAsyncBackendJobExecutionActor?  It looks like the latter could capture a reference to a `Future[Run]`, on which it could then just call `abort()`.